streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -623,16 +630,13 @@ void shutdown(final boolean clean) {
                     }
                 } catch (final TaskMigratedException e) {
                     // just ignore the exception as it doesn't matter during shutdown
-                    task.prepareCloseDirty();
-                    task.closeDirty();
+                    closeTaskDirty(task);
```

Comment:
Ditto, should we close producers as well?

Reply:
Same here, we do close it eventually a few loops after this one